.github/actions/setup_build/action.yml

```diff
@@ -33,12 +33,24 @@ runs:
       shell: bash
       run: |
         pip install lit clang-format==19.1.0 virtualenv
+        # x86: Enable ability to install x86 packages for cross compilation
         if [ "${{ inputs.arch }}" = "x86" ]; then sudo dpkg --add-architecture i386 ; fi
+        # aarch64: Enable ability to install arm64 packages for cross compilation
+        #      The main archive only hosts amd64 and i386, also add ports for arm64.
+        #      Also install Arm/Aarch64 requirements
+        if [ "${{ inputs.arch }}" = "aarch64" ]; then \
+           sudo dpkg --add-architecture arm64 ; \
+           sudo sed -i -e '/^deb /{h;s|deb |&[arch=amd64,i386] |p;g;s|deb http://[^ ]*|deb [arch=arm64] http://ports.ubuntu.com/ubuntu-ports|p;d}' /etc/apt/sources.list; \
+           sudo apt-get install --yes gcc-11-arm-linux-gnueabihf g++-11-arm-linux-gnueabihf; \
+           sudo apt-get install --yes gcc-11-aarch64-linux-gnu g++-11-aarch64-linux-gnu; \
+        fi
         wget -qO - https://packages.lunarg.com/lunarg-signing-key-pub.asc | sudo apt-key add -
         if [ "${{ inputs.ubuntu_version }}" = "20.04" ]; then sudo wget -qO /etc/apt/sources.list.d/lunarg-vulkan-1.3.243-focal.list https://packages.lunarg.com/vulkan/1.3.243/lunarg-vulkan-1.3.243-focal.list; fi
         if [ "${{ inputs.ubuntu_version }}" = "22.04" ]; then sudo wget -qO /etc/apt/sources.list.d/lunarg-vulkan-1.3.243-jammy.list https://packages.lunarg.com/vulkan/1.3.243/lunarg-vulkan-1.3.243-jammy.list; fi
         if [ "${{ inputs.ubuntu_version }}" = "24.04" ]; then sudo wget -qO /etc/apt/sources.list.d/lunarg-vulkan-noble.list https://packages.lunarg.com/vulkan/lunarg-vulkan-noble.list; fi
         sudo apt-get update
+        # Install QEMU for testing cross compilation.
```

Comment:
This is definitely not needed all the time, please add a flag.

Reply:
@coldav  By "flag" do you mean a cross-compilation flag as input to setup_build ??

Reply:
some sort of input, I'm not trying to presuppose what it should be

Reply:
it could be cross_arch: aarch64 for example - which would install qemu and the aarch flags.

Reply:
sorry I was referring to the sed command. I don't think it's needed yet, I think the gcc aarch64  binaries are probably needed for cross compilation.

Reply:
This is now within an `if`